pkg/services/network/network_test.go

```diff
@@ -470,7 +500,124 @@ var _ = Describe("Network provider", func() {
-					"False", apitypes.NamespacedName{Namespace: dummyNs, Name: dummyCluster}, testVnetNotRealizedReason, testVnetNotRealizedMessage)
+					"False", apitypes.NamespacedName{Namespace: dummyNs, Name: dummyCluster}, testNetworkNotRealizedReason, testNetworkNotRealizedMessage)
+				Expect(err).To(MatchError(expectedErrorMessage))
+				Expect(conditions.IsFalse(clusterCtx.VSphereCluster, vmwarev1.ClusterNetworkReadyCondition)).To(BeTrue())
+			})
+		})
+
+		Context("with nsx-vpc network provider and subnetset exists", func() {
+			BeforeEach(func() {
+				client = fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(configmapObj, systemNamespaceObj).Build()
+				vpcNp, _ = NSXTVpcNetworkProvider(client).(*nsxtVPCNetworkProvider)
+				np = vpcNp
+				err = np.ProvisionClusterNetwork(ctx, clusterCtx)
+			})
+
+			It("should not update subnetset", func() {
+				var initialSubnetSetSpec nsxopv1.SubnetSetSpec
+
+				// Fetch the SubnetSet before the operation
+				initialSubnetSet := &nsxopv1.SubnetSet{}
+				err = client.Get(ctx, apitypes.NamespacedName{
+					Name:      dummyCluster,
+					Namespace: dummyNs,
+				}, initialSubnetSet)
+				Expect(err).NotTo(HaveOccurred())
+				initialSubnetSetSpec = initialSubnetSet.Spec
+
+				// Presumably there's code here that might modify the SubnetSet...
+				Expect(err).ToNot(HaveOccurred())
+				subnetset, localerr := np.GetClusterNetworkName(ctx, clusterCtx)
+				Expect(localerr).ToNot(HaveOccurred())
+				Expect(subnetset).To(Equal(clusterCtx.VSphereCluster.Name))
+
+				createdSubnetSet := &nsxopv1.SubnetSet{}
+				err = client.Get(ctx, apitypes.NamespacedName{
+					Name:      dummyCluster,
+					Namespace: dummyNs,
+				}, createdSubnetSet)
+
+				Expect(err).ToNot(HaveOccurred())
+				// Check that the SubnetSetSpec was not changed
+				Expect(createdSubnetSet.Spec).To(Equal(initialSubnetSetSpec), "SubnetSetSpec should not have been modified")
+			})
+
+			It("should successfully retrieve VM service annotations and validate their content and network readiness", func() {
+				annotations, err := np.GetVMServiceAnnotations(ctx, clusterCtx)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(annotations).To(HaveKeyWithValue(NSXTVPCSubnetSetSelectorKey, clusterCtx.VSphereCluster.Name))
+				Expect(conditions.IsTrue(clusterCtx.VSphereCluster, vmwarev1.ClusterNetworkReadyCondition)).To(BeTrue())
+			})
+		})
+
+		Context("with nsx-vpc network provider and subnetset does not exist", func() {
+			var nsxvpcNp *nsxtVPCNetworkProvider
+
+			BeforeEach(func() {
+				client = fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(configmapObj, systemNamespaceObj).Build()
+				nsxvpcNp, _ = NSXTVpcNetworkProvider(client).(*nsxtVPCNetworkProvider)
+				np = nsxvpcNp
+				err = np.ProvisionClusterNetwork(ctx, clusterCtx)
+			})
+
+			It("should create subnetset with new spec", func() {
+				Expect(err).ToNot(HaveOccurred())
```

Comment:
Let's please move this up in l.561

Reply:
I think this one got lost. We should error check after l.571

Reply:
Thanks for reminder.